### DISCUSSION TITLE
What's going on with Node-RED Add-on release?

### DISCUSSION THREAD

**Discussion by alvinchen1** [zachowj/node-red-contrib-home-assistant-websocket#1722] [General]
Absolutely not a criticism and not a whine-session, but am genuinely interested.  Last release was 30-Sept, which was 3 months ago.  The release cadence used to be about once a month.  This [issue](https://github.com/hassio-addons/addon-node-red/issues/2002) was kinda shot down hilariously but with no explanation.

Since then there's been a ton of great PRs, not just the great work from zachowj:

[Node-RED 4.0.3...4.0.8](https://github.com/node-red/node-red/compare/4.0.3...4.0.8)
[websocket 0.73.0...0.75.0](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/compare/v0.73.0...v0.75.0)

I ask because I have paranoia stemming from being heavily invested in SmartThings and WebCORE before I moved to Home Assistant and Node-RED.  I knew I had to start to move away from that platform when I watched both the development decreased or ceased completely which caused concern and the community started to abandon ship.  Migrating platforms is not fun and a lot of work but you gotta balance that with being on a metaphorical island all alone.  While Home Assistant is incredibly vibrant and remains on an upward trajectory, should we be concerned about Node-RED addon support?  There are past examples of addons being archived for various reasons (portainer and pihole etc), migrating away from each of these wouldn't be as bad as trying to migrate away from Node-RED.  With the HA and Node-RED so intertwined, it's hard to see all these Home Assistant updates and no Node-RED addon updates.  Are we going to get to a point where HA charges forward, maybe breaks something and the addon doesn't get fixed?  Should we fork the Node-RED addon and release an unofficial addon?  Again, not a criticism of the devs, they are all incredibly busy and doing amazing things (frenck said on the Home Assistant podcast that he is looking forward to spending time with his family over the holidays and I totally respect that), but NGL am starting to get a little concerned about the longevity of Node-RED since it is 80% of my automations.

Ok, paranoia aside, the Maintenance/License is currently 2024, which means tomorrow, we might see at least an annual update like this [PR](https://github.com/hassio-addons/addon-node-red/pull/1787).

**Comment by zachowj** [comment]
Let me clarify a few things regarding the Node-RED add-on and the `node-red-contrib-home-assistant-websocket` package:

1. **Add-On vs. Package**: The Node-RED add-on for Home Assistant OS (HAOS) is a convenient way for HAOS users to run Node-RED alongside Home Assistant. However, it's important to note that the add-on is not required for using Node-RED or `node-red-contrib-home-assistant-websocket`. You can run Node-RED and the integration completely independently on any platform, without relying on the add-on. 

2. **Control Over the Add-On**: I don’t maintain or have any control over the Node-RED add-on itself or its release cycle. The add-on is maintained as part of the Home Assistant community add-ons, and the best place to track updates or discuss its development is the [Node-RED add-on repository](https://github.com/hassio-addons/addon-node-red). That's the most reliable source for information about when a new version will be released.
<br>As for the issue being closed without a reply, it might have been better suited for the discussions section of the repository rather than the issues section, as it was more of a question than an actual "issue" with the repo.

4. **Updates to `node-red-contrib-home-assistant-websocket`**: While the add-on bundles Node-RED and certain packages like `node-red-contrib-home-assistant-websocket`, the development and release cycle of this package are separate. Updates to `node-red-contrib-home-assistant-websocket` are published regularly and can be installed independently of the add-on by updating your Node-RED palette.

5. **Long-Term Viability**: Node-RED itself is a vibrant and actively maintained project, and `node-red-contrib-home-assistant-websocket` continues to see frequent updates and contributions. There's no indication of development slowing down on either front. The add-on is simply a convenience wrapper for HAOS users, so any concerns about its update cadence shouldn't reflect the health of Node-RED or this integration.

Your point about the add-on being critical for HAOS users who depend on it is valid, and I agree that clarity on its update cycle is valuable. If you're looking for updates or have suggestions for the add-on, engaging directly with the maintainers through the add-on repository is the best approach.

Thanks again for sharing your perspective, and I hope this helps ease some of your concerns!

**Reply by alvinchen1** [reply]
You are totally right.  And thanks for all the development you've done on keeping things going on the websocket package, I really appreciate your responsiveness and understand how critical it is to gluing Home Assistant and Node-RED.  You are 100% right about the vibrancy of Node-RED (I'm an occasional lurker on the forums) outside of the addon and Home Assistant, I didn't really think of that.    

And speaking of the Node-RED addon repository, no sooner had I posted, but I was digging and found that there were some updates going into the [edge repository](https://github.com/hassio-addons/addon-node-red/deployments/edge).  Just being paranoid and too much free time these two week so spending time obsessing over Home Assistant/Node-RED.
